client-keycloak/src/main/java/org/apache/atlas/keycloak/client/service/AtlasKeycloakAuthService.java

```diff
@@ -65,11 +64,11 @@ private OkHttpClient getOkHttpClient() {
     };
 
     public String getAuthToken() {
-        if (Objects.nonNull(currentAccessToken)) {
+        if (!isTokenExpired()) {
             return currentAccessToken.getToken();
         }
         synchronized (this) {
-            if (Objects.isNull(currentAccessToken)) {
+            if (isTokenExpired()) {
```

Comment:
remove if, execution will reach here only if isTokenExpired() == true, no need of checking again

Reply:
there can be multiple threads, which does this checks outside mutex, its like safety net, not to create another token.